File: common/security/src/main/java/com/example/chat/auth/jwt/config/JwtAuthenticationConverterSupport.java
package com.example.chat.auth.jwt.config;

import org.springframework.core.convert.converter.Converter;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.AuthorityUtils;
import org.springframework.security.oauth2.jwt.Jwt;

import java.util.Collection;
import java.util.List;

/**
 * JWT Claims → GrantedAuthority 변환기
 */
public class JwtAuthenticationConverterSupport implements Converter<Jwt, Collection<GrantedAuthority>> {

    @Override
    public Collection<GrantedAuthority> convert(Jwt jwt) {
        List<String> roles = jwt.getClaimAsStringList("roles");
        if (roles == null) {
            return List.of();
        }
        return AuthorityUtils.createAuthorityList(roles);
    }
}

File: common/security/src/main/kotlin/com/example/chat/auth/jwt/config/JwtAuthenticationConverterSupport.kt
package com.example.chat.auth.jwt.config

import org.springframework.core.convert.converter.Converter
import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.AuthorityUtils
import org.springframework.security.oauth2.jwt.Jwt

class JwtAuthenticationConverterSupport : Converter<Jwt, Collection<GrantedAuthority>> {
    override fun convert(jwt: Jwt): Collection<GrantedAuthority> {
        val roles = jwt.getClaimAsStringList("roles") ?: emptyList()
        return AuthorityUtils.createAuthorityList(roles)
    }
}
